### PR TITLE
@W-17376237 Fixing Build Warnings

### PR DESCRIPTION
**Pull request by Crebs**
Xcode build warnings

### CODE REVIEW

File: libs/SalesforceHybridSDK/SalesforceHybridSDK/Classes/SFLocalhostSubstitutionCache.m [hunk 1]
```diff
@@ -34,13 +35,16 @@
 
 @implementation SFLocalhostSubstitutionCache
```
Comment (wmathurin) [inline libs/SalesforceHybridSDK/SalesforceHybridSDK/Classes/SFLocalhostSubstitutionCache.m:36]: Are we still using SFLocalhostSubstitutionCache ??

Comment (Crebs) [reply libs/SalesforceHybridSDK/SalesforceHybridSDK/Classes/SFLocalhostSubstitutionCache.m:36]: Is this something we can remove Wolf?

File: libs/SalesforceHybridSDK/SalesforceHybridSDK/Classes/Plugins/SFSmartStore/SFSmartStorePlugin.m [hunk 2]
```diff
@@ -514,4 +513,8 @@ - (NSString *)internalCursorId:(NSString *) cursorId withGlobal:(BOOL) isGlobal
 - (void)dealloc {
     SFRelease(_cursorCache);
 }
+- (void)pgGetSoupSpec:(nonnull CDVInvokedUrlCommand *)command {
```
Comment (bbirman) [inline libs/SalesforceHybridSDK/SalesforceHybridSDK/Classes/Plugins/SFSmartStore/SFSmartStorePlugin.m:516]: @wmathurin do you think we could get rid of this method in the public header if it hasn't had a definition?

File: libs/SalesforceHybridSDK/SalesforceHybridSDK/Classes/SFLocalhostSubstitutionCache.m (hunk 1 above)
Comment (wmathurin) [reply libs/SalesforceHybridSDK/SalesforceHybridSDK/Classes/SFLocalhostSubstitutionCache.m:36]: I'm not sure yet. Looking through git history.

File: libs/SalesforceHybridSDK/SalesforceHybridSDK/Classes/Plugins/SFSmartStore/SFSmartStorePlugin.m (hunk 2 above)
Comment (wmathurin) [reply libs/SalesforceHybridSDK/SalesforceHybridSDK/Classes/Plugins/SFSmartStore/SFSmartStorePlugin.m:516]: Yes the method can go. We removed it on the javascript side of the plugin already (see https://github.com/forcedotcom/SalesforceMobileSDK-Shared/pull/531/files).

Comment (wmathurin) [reply libs/SalesforceHybridSDK/SalesforceHybridSDK/Classes/Plugins/SFSmartStore/SFSmartStorePlugin.m:516]: I should have removed it in https://github.com/forcedotcom/SalesforceMobileSDK-iOS-Hybrid/pull/211/files

Comment (wmathurin) [reply libs/SalesforceHybridSDK/SalesforceHybridSDK/Classes/Plugins/SFSmartStore/SFSmartStorePlugin.m:516]: Wait I did remove it: https://github.com/forcedotcom/SalesforceMobileSDK-iOS-Hybrid/pull/211/files#diff-515f96aa6001c7aa00d80352293a737c3ef74dbc17fa8f48e1d7cef4e8ae1332L416
Someone must have merged it back.

Comment (wmathurin) [reply libs/SalesforceHybridSDK/SalesforceHybridSDK/Classes/Plugins/SFSmartStore/SFSmartStorePlugin.m:516]: Wait it's coming back with this PR. We shouldn't have that method.

Comment (Crebs) [reply libs/SalesforceHybridSDK/SalesforceHybridSDK/Classes/Plugins/SFSmartStore/SFSmartStorePlugin.m:516]: Ah OK.  I can update this.

**Review by wmathurin — APPROVED** [review]
We can figure out if we need SFLocalhostSubstitutionCache in a future PR.

**Review by bbirman — APPROVED** [review]

**Pull request merged**